boards/arm/stm32h747i_disco/stm32h747i_disco_m7.dts

```diff
@@ -60,6 +60,19 @@
 	status = "okay";
 };
 
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* Set 2KB of storage at the end of 2MB flash */
+		storage_partition: partition@1ff800 {
+			label = "storage";
+			reg = <0x001ff800 0x00000800>;
```

Comment:
Please note that flash is shared between two cores (cf `dts/arm/st/h7/stm32h747Xi_m7.dtsi`)
So the last 1M is assigned to M4 core.
Please put storage at the end of the first 1M.